### PR TITLE
Transport Layer integration tests

### PR DESCRIPTION
**Pull request by sebtomba**
## Overview
Add integration tests for the Transport Layer and `TcpTransportLayer`.

### Which JIRA issue does this PR relate to? If there is not a JIRA issue addressing this work, please create one now and add the link here.
https://rchain.atlassian.net/browse/CORE-607

### Complete this checklist before you submit the PR
- [x] This PR contains no more than 200 lines of code, excluding test code.
- [x] This PR meets [RChain development coding standards](https://rchain.atlassian.net/wiki/spaces/DOC/pages/28082177/Coding+Standards).
- [x] You have someone in mind to assign for review. Make this assignment after submitting the PR.

### Spec output
```
TcpTransportLayer
  when doing a round trip to remote peer
    when everything is fine
    - should send and receive the message
    when response takes to long
    - should fail with a timeout
    when there is no response body
    - should fail with a communication error
    when peer is not listening
    - should fail with peer unavailable error
    when there was a peer-side error
    - should fail with an internal communication error
  when sending a message
  - should deliver the message
  - should not wait for a response
  - should wait for message being delivered (pending)
  when brodacasting a message
  - should send the message to all peers
  when shutting down
    when doing a round trip
    - should not send the message
    when sending a message
    - should not send the message
    when broadcasting a message
    - should not send any messages
```

### CODE REVIEW

**Comment by rabbitonweb** [comment]
Could you possibly printout the output of the spec and add it here to the description of the PR?

File: comm/src/test/scala/coop/rchain/comm/transport/TransportLayerSpec.scala [hunk 1]
```diff
@@ -0,0 +1,251 @@
...
+        "fail with a timeout" in
+          new TwoNodesRuntime[CommErr[Protocol]](Dispatcher.pongDispatcherWithDelay(500)) {
+            def execute(transportLayer: TransportLayer[F],
+                        local: PeerNode,
+                        remote: PeerNode): F[CommErr[Protocol]] =
+              roundTripWithPingAndTimeout(transportLayer, local, remote, 200.millis)
+
+            val result: TwoNodesResult = run()
+
+            result() shouldBe 'left
```
Comment (rabbitonweb) [inline comm/src/test/scala/coop/rchain/comm/transport/TransportLayerSpec.scala]: you could just assert on on `result()  shouldBe Left(Timeout)` ?

File: comm/src/test/scala/coop/rchain/comm/transport/TransportLayerSpec.scala [hunk 2]
```diff
@@ -0,0 +1,251 @@
...
+            val error: CommError = result().left.get
+            error shouldEqual TimeOut
+          }
+      }
+
+      "there is no respon se body" should {
```
Comment (rabbitonweb) [inline comm/src/test/scala/coop/rchain/comm/transport/TransportLayerSpec.scala]: whitespace

File: comm/src/test/scala/coop/rchain/comm/transport/TransportLayerRuntime.scala [hunk 3]
```diff
@@ -0,0 +1,217 @@
...
+
+  def roundTripWithPing(transportLayer: TransportLayer[F],
+                        local: PeerNode,
+                        remote: PeerNode): F[CommErr[Protocol]] =
+    roundTripWithPingAndTimeout(transportLayer, local, remote)
+
+  def roundTripWithPingAndTimeout(transportLayer: TransportLayer[F],
+                                  local: PeerNode,
+                                  remote: PeerNode,
+                                  timeout: FiniteDuration = 3.second): F[CommErr[Protocol]] =
```
[the remark this answers is not in the document]
Comment (sebtomba) [reply comm/src/test/scala/coop/rchain/comm/transport/TransportLayerRuntime.scala]: I needed this for my previous solution where these function has been lifted to a parameter. The compiler was not able to lift a 4 parameter method (with a default value) to a 3 parameter lambda.

File: comm/src/main/scala/coop/rchain/comm/PeerNode.scala [hunk 4]
```diff
@@ -26,20 +26,28 @@ final case class Endpoint(host: String, tcpPort: Int, udpPort: Int) {
 }
 
 // TODO: Add Show instance
-final case class PeerNode(id: NodeIdentifier, endpoint: Endpoint) {
```
[the remark this answers is not in the document]
Comment (sebtomba) [reply comm/src/main/scala/coop/rchain/comm/PeerNode.scala]: reverted

**Pull request merged**